### PR TITLE
Fix style checker bug.

### PR DESCRIPTION
**Pull request by tgamblin**
Fix bug mentioned in #2198 and #1573

### CODE REVIEW

File: lib/spack/spack/cmd/flake8.py [hunk 1]
```diff
@@ -100,6 +102,9 @@ def setup_parser(subparser):
         help="Do not delete temporary directory where flake8 runs. "
              "Use for debugging, to see filtered files.")
 
+    subparser.add_argument(
+        'files', nargs=argparse.REMAINDER, help="specific files to check")
```
Comment (adamjstewart) [inline lib/spack/spack/cmd/flake8.py]: Oh nice!

**Comment by adamjstewart** [comment]
Tried it out and it fixes the URL problem. The new file[s] option works as well! It accepts both relative and absolute paths. 

The only thing I think might confuse users, is if you are not in the `$SPACK_ROOT` directory and run `spack flake8`, it lists modified files relative to the `$SPACK_ROOT`, but you can't use these files in `spack flake8 <files>`.

**Comment by tgamblin** [comment]
@adamjstewart: oh fine.  Happy now?

**Comment by adamjstewart** [comment]
Haha, it was fine before, I was just saying it might confuse someone. I'm literally looking for ways to break it :smile:

**Pull request merged**